### PR TITLE
close all DtlsTransports in pc.close() (including sctp.transport)

### PR DESCRIPTION
**Pull request by aboba**
Fix for Issue https://github.com/w3c/webrtc-pc/issues/1381

Related PR: https://github.com/w3c/webrtc-pc/pull/1414

Rebase of PR https://github.com/w3c/webrtc-pc/pull/1401

***
[Preview](https://s3.amazonaws.com/pr-preview/w3c/webrtc-pc/issue-1381-patchv3.html) | [Diff](https://s3.amazonaws.com/pr-preview/w3c/webrtc-pc/48532fb...f1d3184.html)

### CODE REVIEW

File: webrtc.html [hunk 1]
```diff
@@ -2788,30 +2788,11 @@
...
-                    to <code>"closed"</code>. If
-                    <code><var>receiver.rtcpTransport</var></code> is set, set
-                    <code><var>receiver.rtcpTransport.state</var></code> and
-                    <code><var>receiver.rtcpTransport.transport.state</var></code>
-                    to <code>"closed"</code>.</p>
+                    <p>Set the <code><a data-link-for=
+                    "RTCDtlsTransport">state</a></code> of each of
+                    <var>connection</var>'s <code><a>RTCDtlsTransport</a></code>s
+                    to <code><a data-link-for=
+                    "RTCDtlsTransportState">closed</a></code>.</p>
```
Comment (taylor-b) [inline webrtc.html]: Quotes around "closed"?

Comment (aboba) [reply webrtc.html]: Fixed.

**Comment by fippo** [comment]
You mention "including sctp.transport" in the PR title but it is not in thediff?
But this fixes #1414 afaics 👍

**Comment by aboba** [comment]
@fippo Original issue was that sctp.transport was not closed within the pc.close() steps.  By closing all the DtlsTransport objects (not just those referenced by an RtpSender or RtpReceiver), I believe we have addressed that.

File: webrtc.html [hunk 2]
```diff
@@ -2848,12 +2817,24 @@
                     </ol>
                   </li>
                   <li>
-                    <p>Set <var>connection</var>'s <a>[[\IsClosed]]</a> slot to
-                    <code>true</code>.</p>
+                    <p>Set the <code><a data-link-for=
+                    "RTCDtlsTransport">state</a></code> of each of
```
Comment (jan-ivar) [inline webrtc.html]: [state](https://w3c.github.io/webrtc-pc/#dom-rtcdtlstransport-state) is a read-only attribute and cannot be set.

File: webrtc.html [hunk 3]
```diff
@@ -2848,12 +2817,24 @@
...
-                    <p>Set <var>connection</var>'s <a>signaling state</a> to
-                    <code>"closed"</code>.</p>
+                    <p>Destroy <var>connection</var>'s <a>ICE Agent</a>,
+                    abruptly ending any active ICE processing and
+                    releasing any relevant resources (e.g. TURN
+                    permissions).</p>
+                  </li>
+                  <li>
+                    <p>Set the <code><a data-link-for=
+                    "RTCIceTransport">state</a></code> of each of
```
Comment (jan-ivar) [inline webrtc.html]: [state](https://w3c.github.io/webrtc-pc/#dom-rtcicetransport-state) is a read-only attribute and cannot be set.

Comment (aboba) [reply webrtc.html]: I guess we need a state internal slot...

Comment (aboba) [reply webrtc.html]: Opened Issue https://github.com/w3c/webrtc-pc/issues/1574

**Pull request merged**